dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/DiscoveryRestRepository.java

```diff
@@ -108,6 +110,7 @@ public SearchResultsRest getSearchObjects(final String query, final String dsoTy
 
         } catch (SearchServiceException e) {
             log.error("Error while searching with Discovery", e);
+            throw new IllegalArgumentException();
```

Comment:
Maybe change this to `new IllegalArgumentException("Error while searching with Discovery: " + e.getMessage());` or `new IllegalArgumentException("Error while searching with Discovery, this might be a query syntax exception");` to make clients aware that we use a specific query syntax.

Reply:
Could you also make sure that that message reaches the client (e.g. HAL Browser)?

Reply:
I now also realize that `Warning: (un)escaped characters ahead!` has probably a valid syntax. But it would not return the item with title "Warning: (un)escaped characters ahead!". 

Can you maybe also add an additional test query call for the "Faithful Infidel: Exploring Conformity (2nd edition)" test item but without escaping (thus query `Faithful Infidel: Exploring Conformity (2nd edition)`)? My guess is that that query returns 0 results.